libs/vkd3d/cache.c

```diff
@@ -20,6 +20,104 @@
 
 #include "vkd3d_private.h"
 
+static VkResult vkd3d_create_pipeline_cache(struct d3d12_device *device,
+        size_t size, const void *data, VkPipelineCache *cache)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+
+    VkPipelineCacheCreateInfo info;
+    info.sType = VK_STRUCTURE_TYPE_PIPELINE_CACHE_CREATE_INFO;
+    info.pNext = NULL;
+    info.flags = 0;
+    info.initialDataSize = size;
+    info.pInitialData = data;
+
+    return VK_CALL(vkCreatePipelineCache(device->vk_device, &info, NULL, cache));
+}
+
+#define VKD3D_CACHE_BLOB_VERSION 1
+#define VKD3D_CACHE_BUILD_SIZE 8
+
+struct vkd3d_pipeline_blob
+{
+    uint32_t version;
+    uint32_t vendor_id;
+    uint32_t device_id;
+    char vkd3d_build[VKD3D_CACHE_BUILD_SIZE];
+    uint8_t cache_uuid[VK_UUID_SIZE];
+    char vk_blob[];
+};
+
+HRESULT vkd3d_create_pipeline_cache_from_d3d12_desc(struct d3d12_device *device,
+        const D3D12_CACHED_PIPELINE_STATE *state, VkPipelineCache *cache)
+{
+    const VkPhysicalDeviceProperties *device_properties = &device->device_info.properties2.properties;
+    const struct vkd3d_pipeline_blob *blob = state->pCachedBlob;
+    VkResult vr;
+
+    if (!state->CachedBlobSizeInBytes)
+    {
+        vr = vkd3d_create_pipeline_cache(device, 0, NULL, cache);
+        return hresult_from_vk_result(vr);
+    }
+
+    /* Avoid E_INVALIDARG with an invalid header size, since that may confuse some games */
+    if (state->CachedBlobSizeInBytes < sizeof(*blob) || blob->version != VKD3D_CACHE_BLOB_VERSION)
```

Comment:
Seems like it would be a good idea to use a more specific "magic" number here to avoid any potential mismatch with a D3D12 blob coming from a different implementation.

Reply:
Now using `MAKE_MAGIC(...)` to generate something more specific that's still versioned.